dvc/tree/repo.py

```diff
@@ -306,6 +306,17 @@ def walk_files(self, top, **kwargs):  # pylint: disable=arguments-differ
             for fname in files:
                 yield PathInfo(root) / fname
 
+    def get_dir_hash(self, path_info, **kwargs):
+        if not self.exists(path_info):
+            raise FileNotFoundError
+        _, dvc_tree = self._get_tree_pair(path_info)
+        if dvc_tree and dvc_tree.exists(path_info):
+            try:
+                return dvc_tree.get_dir_hash(path_info, **kwargs)
+            except OutputNotFoundError:
+                pass
+        return super().get_dir_hash(path_info, **kwargs)
```

Comment:
This does not look right. The directory could be mixed as well, i.e. containing both dvc tracked files and git-tracked files.

Reply:
The following could be reproduced with:
```python
def test_import_mixed_dirs(tmp_dir, erepo_dir):
    erepo_dir.gen({"dir": {"foo": "foo", "bar": "bar"}})
    with (erepo_dir / "dir").chdir():
        erepo_dir.dvc.add("foo")

    erepo_dir.scm.add(["dir/foo.dvc", "dir/bar"])
    erepo_dir.scm.commit("add dir")

    from dvc.repo import Repo

    Repo.get(os.fspath(erepo_dir), "dir")
    assert (tmp_dir / "dir").read_text() == {"foo": "foo", "bar": "bar"}
```

Reply:
Thank you @skshetry !